### DISCUSSION TITLE
Build on debian testing (trixie) - a field report

### DISCUSSION THREAD

**Discussion by IdeaHunter** [servo/servo#31016] [Show and tell]
I did found this project in attempt to evaluate if its possible to run a simple browser ui on Pi zero 2 w (512MB of ram) on small screen ~2'' 500x300 amoled display and find an answer about what makes browser eat ~370MB on .html file with just text 'Hello World' in it. In order to find out what eats ram i made a build somewhat working version of servo on my main PC running debian(with cross compilation for now). 

Im leaving here hickups and steps i did as hickup workaround to make somewhat it build as well as reference results, so anybody who is willing to try building on debian would know its possible to build it and would not spend hours to pull it off.

1) First issue i was faced with bootstraping - bootstraping failing at this line
https://github.com/servo/servo/blob/4132bc920b829a82dafae39ff66f2a56c353f46f/python/mach_bootstrap.py#L161C1-L161C78

The python on debian complains that its environment is externally managed and command
```sh
python -m pip install ....
```
would pollute its global packadges and would break system. Given that the all my experience with python is writing sublime plugin and crawl script with usage of virtualenv only once, i choosen an easy route and removed `python -m` effectively making calling it pip directly
![image](https://github.com/servo/servo/assets/10799080/169c92a1-2abc-491a-aa16-3ad0230ceead) and after that it successfully installed requirements

2) Second issue was with bootstraping as well - the script complained it not able to find `libgstreamer-plugins-good1.0-dev`

Upon investingation of debian i found following
There are `libgstreamer-plugins-base1.0-dev` , `libgstreamer-plugins-bad1.0-dev` `libgstreamer-plugins-ugly1.0-dev` and the source package `gst-plugins-good1.0` which is supposed to provide -good dev version https://packages.debian.org/source/trixie/gst-plugins-good1.0 that built from -good repo provides nothing. The hard exerience in past with building libs from sources (0 successfull attempts on more that 10 libs and 4 broken distros) i instantly gave up on idea of building it up and just yeeted it out of dependency list
![image](https://github.com/servo/servo/assets/10799080/6899e119-8b82-4d77-8f61-94d8959e7937)
hoping that 
- either complaition fails at place where its needed and it would me see if -good libs are here or not
- or this dependency is dynamically loaded and would make servo compile and run but blow up at some point under some usecase
- or if this dependency libs provided under debian different package it would just work

This made boostraping happy and let me proceeded to compilation

3) Third issue was with comiling mozangle - it was somehow not happy with float16 being not float described in https://github.com/servo/servo/issues/30587

For which i luckily found workaround at the *last* page *last* link of google
https://zenn.dev/itkrt2y/scraps/7b54c6e9561103

This made it sufficient for finishing dev compiling and running tests/html/about-mozilla.html successfully
The dev build after clean takes 4min 10 sec on laptop Ryzen 7 7745HX with 8 cores and 5.1Ghz boost clock speed which is nice for project of this size 
The release build takes 4min 6 sec

Dev build eats 378 MB of mem and release build eats 352 MB of mem (however value flucuates slightly +-15MB for each run) on tests/html/about-mozilla.htm 

./mach test-unit --release
./mach test-scripts
./mach smoketest --release
Seems all to pass successfully

Can somebody elaborate on problem (2) and tell me how to verify if dependency working or not, maybe there is some test(s) to run or there is some extended test suite which i can ran overnight? (asking out of sheer curiosity and wish to make field report more complete)

**Comment by jdm** [comment]
Based on the list of plugins in https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/tree/discontinued-for-monorepo/ext?ref_type=heads and https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/tree/discontinued-for-monorepo/gst?ref_type=heads you would likely see problems when loading pages that contain audio and video elements. I'm not sure whether that would mean panics/crashes or just failure to play the content.
